stepengine/src/main/java/org/tweetwallfx/devoxx19be/provider/ScheduleDataProvider.java

```diff
@@ -0,0 +1,152 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.tweetwallfx.devoxx19be.provider;
+
+import java.time.LocalDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneId;
+import java.time.format.TextStyle;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import org.tweetwallfx.devoxx.api.cfp.client.CFPClient;
+import org.tweetwallfx.devoxx.api.cfp.client.Schedule;
+import org.tweetwallfx.devoxx.api.cfp.client.ScheduleSlot;
+import org.tweetwallfx.stepengine.api.DataProvider;
+import org.tweetwallfx.stepengine.api.config.StepEngineSettings;
+import static org.tweetwallfx.util.ToString.createToString;
+import static org.tweetwallfx.util.ToString.map;
+
+/**
+ * DataProvider Implementation for Schedule Data
+ */
+public class ScheduleDataProvider implements DataProvider, DataProvider.Scheduled {
+
+    private volatile List<ScheduleSlot> scheduleSlots = Collections.emptyList();
+    private final Config config;
+
+    private ScheduleDataProvider(final Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public ScheduledConfig getScheduleConfig() {
+        return config;
+    }
+
+    @Override
+    public void run() {
+        CFPClient.getClient()
+                .getSchedule(Optional
+                        .ofNullable(System.getProperty("org.tweetwallfx.scheduledata.day"))
+                        .orElseGet(()
+                                -> LocalDateTime.now(ZoneId.systemDefault())
+                                .getDayOfWeek()
+                                .getDisplayName(TextStyle.FULL, Locale.ENGLISH)
+                                .toLowerCase(Locale.ENGLISH)))
+                .map(Schedule::getSlots)
+                .ifPresent(slots -> scheduleSlots = slots);
+    }
+
+    public List<SessionData> getFilteredSessionData() {
+        ZoneId zoneId = Optional.ofNullable(System.getProperty("org.tweetwallfx.scheduledata.zone"))
+                .map(ZoneId::of)
+                .orElseGet(ZoneId::systemDefault);
+        OffsetTime liveOffset = Optional.ofNullable(System.getProperty("org.tweetwallfx.scheduledata.time"))
+                .map(OffsetTime::parse)
+                .orElseGet(() -> OffsetTime.now(zoneId));
+        return SessionData.from(scheduleSlots, liveOffset, zoneId);
+    }
+
+    public static class FactoryImpl implements DataProvider.Factory {
+
+        @Override
+        public DataProvider create(final StepEngineSettings.DataProviderSetting dataProviderSetting) {
+            return new ScheduleDataProvider(dataProviderSetting.getConfig(Config.class));
+        }
+
+        @Override
+        public Class<ScheduleDataProvider> getDataProviderClass() {
+            return ScheduleDataProvider.class;
+        }
+    }
+
+    /**
+     * POJO used to configure {@link TopTalksWeekDataProvider}.
```

Comment:
Wrong link in javadoc
```suggestion
     * POJO used to configure {@link ScheduleDataProvider}.
```